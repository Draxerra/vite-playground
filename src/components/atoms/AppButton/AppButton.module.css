.app-button {
  padding-block: 0.75em;
  padding-inline: 1em;
  border: 0;
  border-radius: 0.5em;
  background-color: var(--clr-primary-500);
  text-transform: uppercase;
  letter-spacing: 0.07em;
  cursor: pointer;
  box-shadow: 0.1em 0.1em 0.2em var(--clr-neutral-900);

  &:hover, &:focus-visible {
    outline: none;
    background-color: var(--clr-primary-300);
  }

  &:focus {
    outline: none;
  }
}
