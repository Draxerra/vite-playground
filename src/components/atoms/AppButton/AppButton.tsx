import React, { FC } from "react";
import classNames from "classnames";

import styles from "./AppButton.module.css";

type AppButtonProps =
  | ({ tag: "a" } & JSX.IntrinsicElements["a"])
  | ({ tag?: "button" } & JSX.IntrinsicElements["button"]);

const AppButton: FC<AppButtonProps> = ({
  children,
  className,
  tag: Component = "button",
  ...props
}) => {
  return (
    <Component
      {...(props as unknown)}
      className={classNames(className, styles.appButton)}
    >
      {children}
    </Component>
  );
};

export default AppButton;
