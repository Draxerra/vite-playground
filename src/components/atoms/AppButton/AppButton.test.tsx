import React from "react";
import AppButton from "./AppButton";
import { render } from "@testing-library/react";

describe("App Button", () => {
  it("should match snapshot", () => {
    const button = render(<AppButton>Test</AppButton>).asFragment();
    expect(button).toMatchSnapshot();
  });
});
