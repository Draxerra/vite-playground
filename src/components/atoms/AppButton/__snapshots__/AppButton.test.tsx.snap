// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Button should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="app-button"
  >
    Test
  </button>
</DocumentFragment>
`;
